core/admin.py

```diff
@@ -37,15 +37,77 @@ def get_queryset(self, request):
     def is_past_event(self, obj):
         return not obj.is_upcoming()
     is_past_event.boolean = True
+    is_past_event.short_description = 'past event?'
 
     def has_stats(self, obj):
         return Postmortem.objects.filter(event=obj).exists()
     has_stats.boolean = True
+    has_stats.short_description = 'has stats?'
+
+    def full_url(self, obj):
+        url = reverse('core:event', kwargs={'city': obj.page_url})
+        url = 'https://djangogirls.org{url}'.format(url=url)
+        return mark_safe('<a href="{url}">{url}</a>'.format(url=url))
+    full_url.short_description = 'page URL'
 
     def get_readonly_fields(self, request, obj=None):
+        fields = set(self.readonly_fields) | {'full_url'}
         if obj and not request.user.is_superuser:
-            return ('email', 'team', 'is_deleted', 'is_on_homepage')
-        return self.readonly_fields
+            fields |= {
+                'city', 'country', 'date', 'email', 'is_on_homepage', 'name', 'page_url', 'team'}
+            # Don't let change objects for events that already happened
+            if not obj.is_upcoming():
+                fields |= {x.name for x in self.model._meta.fields}
+        return fields
+
+    def get_fieldsets(self, request, obj=None):
+        if request.user.is_superuser:
+            return [
+                ('Event info', {'fields': [
+                    'name',
+                    'date',
+                    'city',
+                    'country',
+                    'latlng',
+                    'email',
+                    'page_url',
+                    'is_deleted'
+                ]}),
+                ('Home page', {'fields': [
+                    'photo',
+                    'photo_credit',
+                    'photo_link',
+                    'is_on_homepage'
+                ]}),
+                ('Team', {'fields': [
+                    'main_organizer',
+                    'team'
+                ]}),
+                ('Event page', {'fields': [
+                    'page_title',
+                    'page_description',
+                    'page_main_color',
+                    'page_custom_css',
+                    'is_page_live'
+                ]})
+            ]
+        return [
```

Comment:
Organizers should be able to add / edit homepage photos themselves, but not the `is_on_homepage` parameter.